CMakeLists.txt

```diff
@@ -50,7 +54,10 @@ endif()
 
 if (WIN32)
     install(TARGETS ${PROJECT_NAME}
-        RUNTIME DESTINATION lib)
+        ARCHIVE DESTINATION lib
+        RUNTIME DESTINATION bin
+        PUBLIC_HEADER DESTINATION include/kaitai
```

Comment:
I'm very hesitating about adding it like that. All the best practices normally dictate using `#include <library_name/header_name.h>`, not `#include <header_name.h>` to avoid collisions. Adding `kaitai` directly would promote bad habits and will make it possible to get super weird stuff like `#include "exceptions.h"` randomly including Kaitai Struct exceptions, not your applications one.

Reply:
As I understood things, `PUBLIC_HEADER` in this case only configures where to store headers of the built lib and is not an include dir:

> Any PUBLIC_HEADER files associated with a library are **installed in the destination specified** by the PUBLIC_HEADER argument on non-Apple platforms.[...]

https://cmake.org/cmake/help/latest/command/install.html#targets

So other projects would exactly need to do what you require from my understanding, because the created file path would be `[...]/include/kaitai/header_name.h`:

    #include <kaitai/header_name.h>

Without that statement from my understanding things would result in exactly the behaviour you don't want, because the created file path would be `[...]/include/header_name.h` by default.

Reply:
Got it, thanks for the detailed explanation!